src/roiextractors/imagingextractor.py

```diff
@@ -147,3 +147,57 @@ def write_imaging(imaging, save_path: PathType, overwrite: bool = False):
             If True and save_path is existing, it is overwritten
         """
         raise NotImplementedError
+
+
+class FrameSliceImagingExtractor(ImagingExtractor):
+    """
+    Class to get a lazy frame slice.
+
+    Do not use this class directly but use `recording.frame_slice(...)`
+    """
+
+    extractor_name = "FrameSliceImagingExtractor"
+    installed = True
+    is_writable = True
+    installation_mesg = ""
+
+    def __init__(
+        self, parent_imaging: ImagingExtractor, start_frame: Optional[int] = None, end_frame: Optional[int] = None
+    ):
+        self._parent_imaging = parent_imaging
+        self._start_frame = start_frame
+        self._end_frame = end_frame
+        self._num_frames = self._end_frame - self._start_frame
+
+        parent_size = self._parent_imaging.get_num_frames()
+        if start_frame is None:
+            start_frame = 0
+        else:
+            assert 0 <= start_frame < parent_size
+        if end_frame is None:
+            end_frame = parent_size
+        else:
+            assert 0 < end_frame <= parent_size
+        assert end_frame > start_frame, "'start_frame' must be smaller than 'end_frame'!"
+
+        super().__init__()
+
+    def get_frames(self, frame_idxs: ArrayType, channel: int = 0) -> np.ndarray:
```

Comment:
Type annotation. We should do the same here of `Optional[int]` for future functionality of getting complete frames and not a channel.